atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueServiceTest.java

```diff
@@ -824,6 +832,87 @@ public void testKeyAlreadyExists() {
     }
 
     @Test
+    public void testCheckAndSetFromEmpty() {
+        Assume.assumeTrue(checkAndSetSupported());
+
+        Cell cell = Cell.create(row0, column0);
+        CheckAndSetRequest request = CheckAndSetRequest.newCell(TEST_TABLE, cell, value00);
+        keyValueService.checkAndSet(request);
+
+        verifyCheckAndSet(cell, value00);
+    }
+
+    @Test
+    public void testCheckAndSetFromOtherValue() {
```

Comment:
Could we add one more checkAndSet at the end of this that goes from value01 to value00 and then verifies it again? I'm specifically thinking about Cassandra behavior around lexicographically ordering values on timestamp collisions (which I believe CAS protects us from, but I want to be sure).